apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/grpc/retries/impl/StatusCodesRetryPredicate.java

```diff
@@ -0,0 +1,25 @@
+package io.stargate.sgv2.api.common.grpc.retries.impl;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.stargate.sgv2.api.common.grpc.retries.GrpcRetryPredicate;
+import java.util.List;
+
+/** {@link GrpcRetryPredicate} based on status code matching. */
+public class StatusCodesRetryPredicate implements GrpcRetryPredicate {
+
+  private final List<Status.Code> statusCodes;
```

Comment:
Should be a `Set` since we do lookup for `contains`.

Reply:
Yup makes sense :+1: